### PR TITLE
Better struct doc comments

### PR DESCRIPTION
**Pull request by pepone**
This PR improves the documentation comments for C# generated struct types

### CODE REVIEW

File: cpp/src/slice2cs/Gen.cpp [hunk 1]
```diff
@@ -1750,12 +1749,16 @@ Slice::Gen::TypesVisitor::visitStructStart(const StructPtr& p)
     _out << sb;
 
     _out << sp;
+    _out << nl << "/// <summary>An <see cref=\"ZeroC.Ice.InputStreamReader{T}\"/> used to read <see cref=\""
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1752]:
```suggestion
    _out << nl << "/// <summary>A <see cref=\"ZeroC.Ice.InputStreamReader{T}\"/> used to read <see cref=\""
```

File: cpp/src/slice2cs/Gen.cpp [hunk 2]
```diff
@@ -1750,12 +1749,16 @@ Slice::Gen::TypesVisitor::visitStructStart(const StructPtr& p)
...
+         << name << "\"/> instances.</summary>";
     _out << nl << "public static ZeroC.Ice.InputStreamReader<" << name << "> IceReader =";
     _out.inc();
     _out << nl << "istr => new " << name << "(istr);";
     _out.dec();
 
     _out << sp;
+    _out << nl << "/// <summary>An <see cref=\"ZeroC.Ice.OutputStreamWriter{T}\"/> used to write <see cref=\""
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1760]:
```suggestion
    _out << nl << "/// <summary>A <see cref=\"ZeroC.Ice.OutputStreamWriter{T}\"/> used to write <see cref=\""
```

File: cpp/src/slice2cs/Gen.cpp [hunk 3]
```diff
@@ -1775,10 +1778,19 @@ Slice::Gen::TypesVisitor::visitStructEnd(const StructPtr& p)
 
     if(partialInitialize)
     {
-        _out << sp << nl << "partial void Initialize();";
+        _out << sp;
+        _out << nl << "/// <summary>The generated constructor calls the Initialize partial method after initializing "
+             << "the struct data members.</summary>";
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1783]:
```suggestion
        _out << nl << "/// <summary>The constructor calls the Initialize partial method after initializing "
             << "the data members.</summary>";
```

File: cpp/src/slice2cs/Gen.cpp [hunk 4]
```diff
@@ -1775,10 +1778,19 @@ Slice::Gen::TypesVisitor::visitStructEnd(const StructPtr& p)
...
+        _out << nl << "partial void Initialize();";
     }
 
     _out << sp;
+    _out << nl << "/// <summary>Constructs a new instance of the <see cref=\"" << name << "\"/> struct type.</summary>";
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1788]: Is it really necessary to say `... struct type.`?
```suggestion
    _out << nl << "/// <summary>Constructs a new instance of <see cref=\"" << name << "\"/> .</summary>";
```

File: cpp/src/slice2cs/Gen.cpp [hunk 5]
```diff
@@ -1802,6 +1814,9 @@ Slice::Gen::TypesVisitor::visitStructEnd(const StructPtr& p)
     _out << eb;
 
     _out << sp;
+    _out << nl << "/// <summary>Constructs a new instance of the <see cref=\"" << name << "\"/> struct type.</summary>";
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1817]: Ditto?
```suggestion
    _out << nl << "/// <summary>Constructs a new instance of <see cref=\"" << name << "\"/>.</summary>";
```

File: cpp/src/slice2cs/Gen.cpp [hunk 6]
```diff
@@ -1900,6 +1931,7 @@ Slice::Gen::TypesVisitor::visitEnum(const EnumPtr& p)
     emitDeprecate(p, 0, _out, "type");
     emitCommonAttributes();
     emitCustomAttributes(p);
+    writeTypeDocComment(p, "");
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1934]: What does `""` mean here?

File: cpp/src/slice2cs/Gen.cpp [hunk 7]
```diff
@@ -1923,6 +1955,7 @@ Slice::Gen::TypesVisitor::visitEnum(const EnumPtr& p)
 
     _out << sp;
     emitCommonAttributes();
+    _out << nl << "/// <summary>Helper class for marshal and unmarshal <see cref=\"" << name << "\"/>.</summary>";
```
Comment (externl) [inline cpp/src/slice2cs/Gen.cpp:1958]:
```suggestion
    _out << nl << "/// <summary>Helper class for marshaling and unmarshaling <see cref=\"" << name << "\"/>.</summary>";
```

**Review by externl — APPROVED** [review]

File: cpp/src/slice2cs/Gen.cpp (hunk 6 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp]: Fixed, it's the deprecated reason if the type was deprecated.

**Pull request merged**